templates/install.j2

```diff
@@ -119,75 +184,35 @@ yunohost service add $app --log="/var/log/$app/$app.log"
 {% endif %}
 
 {% if data.use_logrotate %}
-# Use logrotate to manage application logfile(s)
-ynh_use_logrotate
+ynh_config_add_logrotate
 {% endif %}
 
 {% if data.use_fail2ban -%}
-# Create a dedicated Fail2Ban config
-ynh_add_fail2ban_config --logpath="/var/log/nginx/${domain}-error.log" --failregex="{{ data.fail2ban_regex }}"
+ynh_config_add_fail2ban --logpath="/var/log/nginx/${domain}-error.log" --failregex="{{ data.fail2ban_regex }}"
 {% endif %}
 
 {% if data.use_cron -%}
-# Add cron job
 cron_path="/etc/cron.d/$app"
-ynh_add_config --template="../conf/task.cron" --destination="$cron_path"
+ynh_config_add --template="../conf/task.cron" --destination="$cron_path"
 chown root: "$cron_path"
 chmod 644 "$cron_path"
 {% endif %}
 
-#=================================================
-# APP INITIAL CONFIGURATION
-#=================================================
-# ADD A CONFIGURATION
-#=================================================
-ynh_script_progression --message="Adding app's configuration file..."
-
-{% if data.generator_mode == 'tutorial' -%}
-### You can add specific configuration files.
-###
-### Typically, put your template conf file in ../conf/your_config_file
-### The template may contain strings such as __FOO__ or __FOO_BAR__,
-### which will automatically be replaced by the values of $foo and $foo_bar
-###
-### ynh_add_config will also keep track of the config file's checksum,
-### which later during upgrade may allow to automatically backup the config file
-### if it's found that the file was manually modified
-###
-### Check the documentation of `ynh_add_config` for more info.
-{% endif %}
-
-ynh_add_config --template="{{ data.custom_config_file }}" --destination="$install_dir/{{ data.custom_config_file }}"
-
-# FIXME: this should be handled by the core in the future
-# You may need to use chmod 600 instead of 400,
-# for example if the app is expected to be able to modify its own config
-chmod 400 "$install_dir/{{ data.custom_config_file }}"
-chown $app:$app "$install_dir/{{ data.custom_config_file }}"
-
-{% if data.generator_mode == 'tutorial' -%}
-### For more complex cases where you want to replace stuff using regexes,
-### you shoud rely on ynh_replace_string (which is basically a wrapper for sed)
-### When doing so, you also need to manually call ynh_store_file_checksum
-###
-### ynh_replace_string --match_string="match_string" --replace_string="replace_string" --target_file="$install_dir/some_config_file"
-### ynh_store_file_checksum --file="$install_dir/some_config_file"
-{% endif %}
-
 {% if data.install_snippet -%}
 #=================================================
 # INSTALL APP
 #=================================================
-ynh_script_progression --message="Installing app..." --weight=5
+ynh_script_progression "Installing $app..."
 
 pushd $install_dir
-{{ data.install_snippet }}
+  ynh_hide_warnings {{ data.install_snippet }}
```

Comment:
I would discourage using `ynh_hide_warnings` by default because it may contain important information ... (and in the past the npm/node call were kind of annoying but if I remember correctly I improved the situation such that it should be less necessary to hide warning)

Also the `install_snippet` may in fact be multiline